### DISCUSSION TITLE
Consuming scoped dependencies from commands?

### DISCUSSION THREAD

**Discussion by WurliMonkhaven** [spectreconsole/spectre.console#1377] [Q&A] [answered]
Hi folks. I'm using .NET's generic host stuff to handle my DI. I was wondering if anyone knew a way I could register and consume scoped dependencies from commands? I have a large enterprise app I'm trying to hack this to work with and it has some scoped dependencies.

**Comment by nils-a** [comment, marked as answer]
Since we do not support DI any containers directly, but rather enable adapters for DI containers, this would entirely depend on the adapter you are using.

**Reply by WurliMonkhaven** [reply]
Could you explain more? I'm not sure what you mean by adapter in this context.

**Reply by nils-a** [reply]
Spectre.Console does not provide any specific DI container, but rather you'll need to provide an "adapter" to your specific solution.
Have a look at the "Dependency Injection" section at https://spectreconsole.net/cli/commandapp.
